python/feldera/pipeline.py

```diff
@@ -221,35 +287,51 @@ def wait_for_completion(self, shutdown: bool = False):
         if shutdown:
             self.shutdown()
 
-    def start(self):
+    def start(self, timeout_s: Optional[float] = None):
         """
         .. _start:
 
         Starts this pipeline.
 
-        :raises RuntimeError: If the pipeline returns unknown metrics.
+        The pipeline must be in SHUTDOWN state to start.
+        If the pipeline is in any other state, an error will be raised.
+        If the pipeline is in PAUSED state, use `.meth:resume` instead.
+        If the pipeline is in FAILED state, it must be shutdown before starting it again.
+
+        :param timeout_s: The maximum time (in seconds) to wait for the pipeline to start.
+
+        :raises RuntimeError: If the pipeline is not in SHUTDOWN state.
         """
 
         status = self.status()
         if status != PipelineStatus.SHUTDOWN:
+            deployment_error = self.client.get_pipeline(self.name).deployment_error
+            error_msg = ""
+            if deployment_error is not None:
+                error_msg = deployment_error.get("message", "")
+
             raise RuntimeError(
                 f"pipeline {self.name} in state {str(status.name)} cannot be started\n"
-                + self.client.get_pipeline(self.name).deployment_error.get(
-                    "message", ""
-                )
+                + error_msg
```

Comment:
I think you want different messages depending on whether the pipeline is in a failed state.

If the pipeline is _not_ in failed state:

"Cannot start pipeline XXX in state YYY. The pipeline must be in the SHUTDOWN state before it can be started. You can either shut down the pipeline using the ZZZ method or use AAA to resume a paused pipeline."

If the pipeline _is_ failed:

"Cannot start pipeline XXX in FAILED state. The pipeline must be in the SHUTDOWN state before it can be started, but it is currently in the FAILED state (error message: EEE). Use ZZZ method to shut down the pipeline."